lib/internal/modules/esm/loader.js

```diff
@@ -70,28 +87,30 @@ class ESMLoader {
   /**
    * Prior to ESM loading. These are called once before any modules are started.
    * @private
-   * @property {Function[]} globalPreloaders First-in-first-out list of
-   * preload hooks.
+   * @property {Map<URL['href'], Function>} globalPreloaders Last-in-first-out
+   *  list of preload hooks.
    */
-  #globalPreloaders = [];
+  #globalPreloaders = new SafeMap();
 
   /**
    * Phase 2 of 2 in ESM loading.
    * @private
-   * @property {Function[]} loaders First-in-first-out list of loader hooks.
+   * @property {Map<URL['href'], Function>} loaders Last-in-first-out
+   *  collection of loader hooks.
    */
-  #loaders = [
-    defaultLoad,
-  ];
+  #loaders = new SafeMap([
+    ['node:esm/load.js', defaultLoad],
```

Comment:
I considered that, but the stack-trace is dynamically generated, and this is static, so if the file got moved, someone would need to know to update this path. I did this short version because it seemed to me that the true full path doesn't actually matter (it just needs to indicate that it's Node's internal one).

Not opposed to making it the full path if preferred.

Reply:
It can get outputted in a stack-trace, so let's make it the real path :)